### ISSUE TITLE
sql: modification_time field not filled in when a table is created

### ISSUE THREAD

**Issue by a-robinson** [cockroachdb/cockroach#9188]
I have to admit that I'm not sure whether this is actually a bug or if `modification_time` is only meant to populated when a table is modified after creation, but I found it surprising that when creating a table the `TableDescriptor`'s `modification_time` is persisted to the store with a zero value.

For example, this is what one such `TableDescriptor` proto looks like after being created:

```
name:"accounts2" id:53 parent_id:50 version:1 up_version:false modification_time:<wall_time:0 logical:0 > columns:<name:"id" id:1 type:<kind:INT width:0 precision:0 > nullable:false hidden:false > columns:<name:"balance" id:2 type:<kind:DECIMAL width:0 precision:0 > nullable:true hidden:false > next_column_id:3 families:<name:"primary" id:0 column_names:"id" column_names:"balance" column_ids:1 column_ids:2 default_column_id:2 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true column_names:"id" column_directions:ASC column_ids:1 foreign_key:<table:0 index:0 name:"" > interleave:<> > next_index_id:2 privileges:<users:<user:"root" privileges:2 > > next_mutation_id:1 format_version:FamilyFormatVersion state:PUBLIC view_query:""
```

Is it intentional that `modification_time` is 0 or just an accident?

**Comment by knz** [comment]
I'd be ok with prefilling that field with the same time the table was created.

**Comment by knz** [comment]
This is a schema change issue, cc @vivekmenezes for triage and prioritization.

**Comment by knz** [comment]
& @jordanlewis

**Comment by lego** [comment]
Closing as it looks like this is has already been addressed by 78fded4d1307a08f154cc9d336cdaf80fe88aa74. This was a problem!

**Comment by vivekmenezes** [comment]
and https://github.com/cockroachdb/cockroach/issues/17526
